### PR TITLE
[wifi-login-pages] Added remember me checkbox #63

### LINKED ISSUE #63: [wifi-login-pages] Add remember me checkbox
Add a "remember me" checkbox, enabled by deafult.

If the checkbox is disabled, the app will not store session information in cookies (as it does now by default), but it will use `sessionStorage` as an alternative way to recognize that users have already logged-in (showing them the status page).

This feature is important because public wifi is sometimes used in shared computers in public places, and therefore we may have to disable it by default.

### PR DESCRIPTION
**Pull request by codesankalp**
Added remember_me checkbox with sessionStorage.

Closes #63

### CODE REVIEW

**Comment by codesankalp** [comment]
@nemesisdesign, @niteshsinha17 Can you review it?
I tried to implement the sessionStorage when `remember_me` is unchecked and made it compatible with all other previous code.

**Comment by niteshsinha17** [comment]
@codesankalp You have formatted lots of files. It made it difficult to review also to understand which change is made for this particular issue.

**Comment by codesankalp** [comment]
@niteshsinha17 I will unstage the reformatted file in which I haven't added a code for this implementation.

**Comment by codesankalp** [comment]
@niteshsinha17 I have removed the extra reformatted files.

Also, I have implemented the remember_me feature only for the login.
If it's required for registration then do let me know.

**Comment by codesankalp** [comment]
@nemesisdesign Can you please review it?
I am using commit of this PR in https://github.com/openwisp/openwisp-wifi-login-pages/issues/171.
And will submit a PR soon.

**Comment by nemesisdesign** [comment]
@codesankalp sorry, I need more days.

File: client/components/login/login.js [hunk 1]
```diff
@@ -59,8 +60,8 @@ export default class Login extends React.Component {
   }
 
   getEmailField = (input_fields) => {
-    const {email, errors} = this.state;
-    const {language} = this.props;
+    const { email, errors } = this.state;
+    const { language } = this.props;
```
Comment (nemesisdesign) [inline client/components/login/login.js:64]: please avoid these kind of style changes. If you want to add an automatic formatting tool that does this kind of change, please open a new issue and a new PR just for that.

This applies to any change in this PR which is not directly related to the feature you're working on.

File: internals/generators/organization/config.yml.hbs [hunk 2]
```diff
@@ -422,6 +422,10 @@ client:
             en: "password"
           label:
             en: "password"
+        remember_me:
+          type: "checkbox"
+          label:
+            en: "Remember Me"
```
Comment (nemesisdesign) [inline internals/generators/organization/config.yml.hbs:428]: let's change this to "Automatically log me in to the WiFi service at the next visit" which I hope most end-users will be able to understand

File: org-configurations/default-configuration.yml [hunk 3]
```diff
@@ -159,7 +159,7 @@ client:
           pattern_description:
             en: "only letters, numbers, and @/./+/-/_ characters"
           placeholder:
-            en: "enter location" 
+            en: "enter location"
```
Comment (nemesisdesign) [inline org-configurations/default-configuration.yml:162]: avoid this unrelated change please

File: client/components/login/login.test.js [hunk 4]
```diff
@@ -110,6 +110,15 @@ describe("<Login /> interactions", () => {
...
 
+  it("should change state value when handleCheckBoxChange function is invoked", () => {
+    // phone_number should not be present if mobile_phone_verification is off
+    expect(wrapper.find("[name='phone_number']")).toEqual({});
+    wrapper
+      .find("#remember_me")
+      .simulate("change", { target: { checked: false, name: "remember_me" } });
+    expect(wrapper.state("remember_me")).toEqual(false);
+  });
+
```
Comment (nemesisdesign) [inline client/components/login/login.test.js:121]: Good that you added one test, although looks not sufficient to prevent this feature from breaking.
Is there any measure you can include in this PR so that if someone changes the code and breaks this feature, the tests will find out?

**Review by nemesisdesign — CHANGES_REQUESTED** [review]
Sorry for my delay @codesankalp, here's my review.

First of all, thank you for your effort, I tested it and it works! 
We need to improve it a bit, see below.

### Remember me choice should be maintained

If I choose to opt out from "remember me" and I close and then open the browser again, the checkbox should remain unchecked because otherwise the user could inadvertently log-in with the option activated from a shared computer. Can you please do this change (we can use localStorage)?

### Configurable default value

Can you make the default value of the checkbox configurable?
Some orgs may want to have this turned on by default while others off by default.

**Comment by codesankalp** [comment]
Sorry for the delay @nemesisdesign Most of the work is done but needs some more time in adding tests for session storage so that this feature can be tested well and it will not break again in the future.

**Comment by codesankalp** [comment]
@nemesisdesign Needs your review.

File: internals/generators/organization/index.js [hunk 5]
```diff
@@ -85,6 +85,12 @@ module.exports = {
       message: "Does your captive portal support log out by session ID?",
       default: false,
     },
+    {
+      type: "confirm",
+      name: "remember_me",
+      message: "Does your captive portal support automatically log me in?",
```
Comment (nemesisdesign) [inline internals/generators/organization/index.js:91]: mm.. did I ask you to add this?

If I am not missing anything, this is not a feature of the captive portal but something we are adding on top, because we save the session of the user from which we're then able to retrieve the radius user token and log in the user automatically.

So the question should be something like: `Do you want to support automatic logins (remember me checkbox)?`

File: org-configurations/default-configuration.yml [hunk 6]
```diff
@@ -340,6 +340,11 @@ client:
             en: "password"
           label:
             en: "password"
+        remember_me:
+          type: "checkbox"
+          value: true
+          label:
+            en: "Automatically log me in to the WiFi service at the next visit"
```
Comment (nemesisdesign) [inline org-configurations/default-configuration.yml:347]: after testing it I noticed it's too long, can you please shorten this and the other occurrences to something more conservative like: `Remember me on this device`?

**Review by nemesisdesign — CHANGES_REQUESTED** [review]
Thank you @codesankalp, this is almost ready, see below for my final requests for improvement.

**Comment by codesankalp** [comment]
@nemesisdesign Should I squash the commits?

**Comment by nemesisdesign** [comment]
> @nemesisdesign Should I squash the commits?

No, don't worry, I'll do it.

**Review by nemesisdesign — APPROVED** [review]
Thanks! :+1:
